src/seahawk/seahawk_deck/thrust.py

```diff
@@ -39,28 +44,223 @@ class Thrust(Node):
+
+        self.motor_thrusts = [
+            [0.0, 0.0, 1.0],
+            [0.0, 0.0, 1.0],
+            [0.0, 0.0, 1.0],
+            [0.0, 0.0, 1.0],
+            [0.7071, -0.7071, 0],
+            [0.7071, 0.7071, 0],
+            [-0.7071, -0.7071, 0],
+            [-0.7071, 0.7071, 0]
+        ]
+
+        self.declare_parameter('center_of_mass_offset', [0.0, 0.0, 0.0])
+        self.center_of_mass_offset = self.get_parameter('center_of_mass_offset').value
+
+        self.add_on_set_parameters_callback(self.update_center_of_mass)
+
+        self.motor_config = self.generate_motor_config()
+        self.inverse_config = np.linalg.pinv(self.motor_config, rcond=1e-15, hermitian=False)
+
         self.subscription = self.create_subscription(Twist, 'drive/twist', self._callback, 10)
         self.motor_pub = self.create_publisher(Float32MultiArray, 'drive/motors', 10)
-        
-
-    # Calculates what a thruster should output based on multiple input values
-    def combine_input(self, direction1:float, direction2:float)->float:
-        """Add two directions in such a way that they do not fall outside [-1, 1]"""
-        if (direction1 >= 0 and direction2 >= 0): # If both input values are positive (0 included)
-            # Combines decimal percentage values based on a probability union operation to determine what the thruster should output
-            return direction1 + direction2 - (direction1 * direction2)
-        elif (direction1 < 0 and direction2 < 0): # If both input values are negative
-            # Tweaked probability union operation similar to above to work with negative values
-            return direction1 + direction2 + (direction1 * direction2)
-        else:
-            # If one value is positive and one value is negative, adds the values of different signs to offset each other
-            return direction1 + direction2
+        self.__params = Thrust.__generate_curve_fit_params()
+
+    def get_polynomial_coef(self, mv: list, limit: float) -> list:
+        """
+        Args:
+            mv: The motor values in newtons that when produced will result in our desired twist
+            limit: The current limit we would like to stay under in amperes (TOTAL_CURRENT_LIMIT or ESC_CURRENT_LIMIT)
+
+        Returns:
+            A list of the coefficients of a 5th degree polynomial function, where the input of said
+            function is the scaling factor and the output is the current (A) draw
+        """
+        return [self.__params[0] * sum([thrust**6 for thrust in mv]),
+                self.__params[1] * sum([thrust**5 for thrust in mv]),
+                self.__params[2] * sum([thrust**4 for thrust in mv]),
+                self.__params[3] * sum([thrust**3 for thrust in mv]),
+                self.__params[4] * sum([thrust**2 for thrust in mv]),
+                self.__params[5] * sum(mv),
+                self.__params[6] * len(mv) - limit]
+
+    def get_current_scalar_value(self, mv: list, limit: float) -> float:
```

Comment:
Consider adding a link to the paper you wrote on the methodology. Add the pdf to the [doc](https://github.com/CabrilloRoboticsClub/cabrillo_rov_2023/tree/main/doc) directory

Reply:
Moved to Issue #194